public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts

```diff
@@ -225,6 +225,33 @@ describe('AzureLogAnalyticsDatasource', () => {
         expect(results[1].value).toBe('Policy');
       });
     });
+
+    describe('and contain options', () => {
+      const queryResponse = {
+        tables: [],
+      };
+
+      beforeEach(async () => {
+        ctx.ds.azureLogAnalyticsDatasource.getResource = jest.fn().mockImplementation((path: string) => {
+          const params = new URLSearchParams(path.split('?')[1]);
+          const query = params.get('query');
+          expect(query).toEqual(
```

Comment:
I don't love that the expect is in the beforeEach because there's no guarantee that this `getResource` function is called by `metricFindQuery`

But I also see that we do this pattern elsewhere so it probably is not worth changing right now, maybe in the future we can think of another way to do it.

Reply:
right, there is actually no reason to do this in a `beforeEach` hook, I will move that to the test.